pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -138,6 +150,39 @@ func (c *Controller) finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 	return reconcile.Result{}, nil
 }
 
+func (c *Controller) ensureTerminationGracePeriodTerminationTimeAnnotation(ctx context.Context, node *v1.Node, nodeClaim *v1beta1.NodeClaim) error {
+	// if the expiration annotation is already set, we don't need to do anything
+	if _, exists := node.ObjectMeta.Annotations[v1beta1.NodeTerminationTimestampAnnotationKey]; exists {
+		return nil
+	}
+
+	// Unlike Pods, the default and un-changeable setting of the a NodeClaim's terminationGracePeriodSeconds is 0, meaning the DeletionTimestamp is always equal to the time the NodeClaim is deleted.
+	if nodeClaim.Spec.TerminationGracePeriod != nil && !nodeClaim.ObjectMeta.DeletionTimestamp.IsZero() {
+		terminationTimeString := nodeClaim.DeletionTimestamp.Time.Add(nodeClaim.Spec.TerminationGracePeriod.Duration).Format(time.RFC3339)
```

Comment:
It will be annotated in the same timezone as the node where the Karpenter controller is running. Apparently this is a source of contention for some golang users.

https://groups.google.com/g/golang-nuts/c/dyPTdi6oem8?pli=1

Do we want to force all timestamps for this field to be in UTC if users are running Karpenter on other timezones? Are you ok with me using `.UTC()` to ensure these time values are in UTC, but losing the monotonic clock benefits (seems like some small performance benefits when comparing Time objects)?

Reply:
This is a good point. I could understand that from Karpenter's perspective, we'll be able to parse whatever the timezone is, and that this isn't manually set by the user. I'm not grokking the monotonic clock benefits, but actually thinking about it more, I'm happy to keep this as is.